### PR TITLE
fix: `message_type` inconsistency across message end points

### PR DESCRIPTION
**Pull request by muhsin-k**
The `before_type_cast` method sometimes returns a string for `message_type`, creating inconsistencies in different payloads. This pull request will remove all `before_type_cast` usage and replace it with `to_i` methods.

### CODE REVIEW

File: spec/controllers/api/v1/widget/conversations_controller_spec.rb [hunk 1]
```diff
@@ -69,9 +69,9 @@
       expect(json_response['id']).not_to be_nil
       expect(json_response['contact']['email']).to eq 'contact-email@chatwoot.com'
       expect(json_response['contact']['phone_number']).to eq '+919745313456'
-      expect(json_response['contact']['name']).to eq 'contact-name'
```
Comment (sojan-official) [inline spec/controllers/api/v1/widget/conversations_controller_spec.rb:72]: itentional?

File: app/views/api/v1/accounts/search/_message.json.jbuilder [hunk 2]
```diff
@@ -1,6 +1,6 @@
 json.id message.id
 json.content message.content
-json.message_type message.message_type_before_type_cast
+json.message_type message.message_type.to_i
```
Comment (sojan-official) [inline app/views/api/v1/accounts/search/_message.json.jbuilder:3]: lets also add a spec for the search controller message payload

**Review by sojan-official — COMMENTED** [review]
looks mostly good. added couple of comments for missing specs

File: spec/controllers/api/v1/widget/conversations_controller_spec.rb [hunk 3]
```diff
@@ -70,8 +74,19 @@
       expect(json_response['contact']['email']).to eq 'contact-email@chatwoot.com'
       expect(json_response['contact']['phone_number']).to eq '+919745313456'
       expect(json_response['contact']['name']).to eq 'contact-name'
+    end
+
+    it 'creates a conversation with correct message and custom attributes' do
```
Comment (muhsin-k) [inline spec/controllers/api/v1/widget/conversations_controller_spec.rb:79]: Added another spec to address the rubocop warning `RSpec/MultipleExpectations`

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**